docs/how-to-guides/admin/gitlab.rst

```diff
@@ -0,0 +1,240 @@
+`official GitLab cloud-native Helm chart
+<https://docs.gitlab.com/charts/>`_.
+
+External GitLab
+---------------
+
+This section explains how to link Renku with an existing GitLab deployment.
+
+External GitLab Setup
+^^^^^^^^^^^^^^^^^^^^^
+
+Renku needs to authenticate users with the Gitlab instance it is using.
+Depending on whether you have admin access to the external GitLab instance, 
+the following options are available to you:
+
+  * If you *do not* have admin access to GitLab you can:
+     * Use GitLab as an identity provider for Renku
+     * Have separate Renku and GitLab identities
+   
+  * If you *do* have admin access to GitLab you can:
+     * Use GitLab as an identity provider for Renku
+     * Use Renku's Keycloak as an identity provider for GitLab
+     * Have separate Renku and GitLab identities
+
+To use an external GitLab as a backend for a Renku deployment you must register
+the Renku application in GitLab. If you are not an admin user, this is done under
+your personal user preferences. It is possible that the GitLab instance you are
+using prevents users from registering external applications - in this case you
+must contact your GitLab server administrator for assistance.
+
+Renku GitLab application details
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Name: renku
+
+Callback URLs:
+
+.. code-block:: console
+
+  https://<your-renku-dns>/login/redirect/gitlab
+  https://<your-renku-dns>/api/auth/gitlab/token
+
+Scopes:
+
+.. code-block:: console
+
+    api (Access the authenticated user's API)
+    read_user (Read the authenticated user's personal information)
+    read_repository (Allows read-access to the repository)
+    read_registry (Grants permission to read container registry images)
+    openid (Authenticate using OpenID Connect)
+
+Copy the secret and client ID for use in the step below.
+
+Configuring an External GitLab as Identity Provider for Renku
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the Renku deployment is relying on an external GitLab instance, 
+this GitLab instance could be configured as an identity provider for Renku's Keycloak. 
+This way the Renku login screen will show the option to "Login via GitLab" 
+and existing GitLab users can use Renku without creating a separate Renku account.
+
+#. Login to the admin console of Renku's Keycloak, usually found at \https://<renku-domain>/auth, using the username and password stored in the Kubernetes secret called  ``keycloak-password-secret``. Ensure you ``base64`` decode these values before using them to login. If you use an external Keycloak with Renku, login to the external Keycloak instead.
+#. Add an ``Identity Provider`` of type ``OpenID Connect v1.0``.
+#. Set ``Alias`` to <renku-domain>, ``Authorization URL`` to \https://<gitlab-domain>/oauth/authorize, ``Token URL`` to \https://<gitlab-domain>/oauth/token and ``Client ID`` and ``Client Secret`` to the respective values copied from the Renku GitLab application details.
```

Comment:
Could these be bullet points? Also, there are 2 things missing which I needed to specify:

![image](https://user-images.githubusercontent.com/540360/202716673-786a7342-9d34-407e-993d-600add12af51.png)

![image](https://user-images.githubusercontent.com/540360/202716766-c31aaa43-1e03-4621-859e-738440ce2a14.png)

(I'm not sure if I picked the correct values...)

Reply:
@seanrmurphy I am not sure if gitlab supports client secret sent as JWT. I have always done it with "client secret sent as post". When you select "client secret sent as post" then you do not have to pick a signature algorithm.

Sending the secret as JWT is definitely safer but I am not sure if this is possible from the Gitlab side. @wesjdj did you see anything about this in the helm chart setup? We should try this and ensure it works if we want to put it in the docs. You can easily add this option to a running dev deployment and check if you can login via gitlab.

Reply:
Either way we should be clear which client authentication method should be used. Or at least recommend one as preferred.

Reply:
@seanrmurphy if you use the `client secret sent as post` method for `Client Authentication` then you do not have to select a signature algorithm. That is the only one I have used when I have set this up for my dev deployments. Sending the secret as JWT is safer but I am not sure if it will work.

If we want to add the JWT option to the docs we should try it first to make sure it works. You can easily try if the JWT method will work on a ci dev deployment.

Anyhow we should definitely specify in the docs, right in this location which `Client Authentication` method should be used.